Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbApplication.cpp

```diff
@@ -17,25 +18,24 @@ namespace AzFramework
     {
     public:
         XcbConnectionManagerImpl()
+            : m_xcbConnection(xcb_connect(nullptr, nullptr))
         {
-            m_xcbConnection = xcb_connect(nullptr, nullptr);
             AZ_Error("Application", m_xcbConnection != nullptr, "Unable to connect to X11 Server.");
             XcbConnectionManagerBus::Handler::BusConnect();
         }
 
         ~XcbConnectionManagerImpl() override
         {
             XcbConnectionManagerBus::Handler::BusDisconnect();
-            xcb_disconnect(m_xcbConnection);   
         }
 
         xcb_connection_t* GetXcbConnection() const override
         {
-            return m_xcbConnection;
+            return m_xcbConnection.get();
         }
 
     private:
-        xcb_connection_t*   m_xcbConnection = nullptr;
+        XcbUniquePtr<xcb_connection_t, xcb_disconnect> m_xcbConnection = nullptr;
```

Comment:
Where is 'XcbUniquePtr' coming from? I dont see it in this PR

Reply:
It was added to `Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInterface.h` in f5effaabccc. At the time I added that file, I was planning on having to add a pure virtual interface that the production code would use, and the production code would provide a concrete implementation of, that would talk to the real `libxcb`. But with this approach, we can get the production code to talk directly to `libxcb`, and still write tests against a mock implementation (which is why this PR adds a `MockXcbInterface.h`). So maybe we should rename that `XcbInterface.h` file now.